### ISSUE TITLE
look forward to publishing pre-trained models

### ISSUE THREAD

**Issue by shisi-cc** [SaberaTalukder/TOTEM#3]
I am very interested in this work and look forward to publishing pre-trained models in huggingface repository.

**Comment by Antho2422** [comment]
+1 ! I would love to test this work on EEG data for my PhD

**Comment by j-adamczyk** [comment]
+1, this would be very useful for me also

**Comment by SaberaTalukder** [comment]
- Pretrained generalist tokenizers: https://drive.google.com/drive/u/0/folders/1TSwPHDMAhcpe2AKl4xsVbUUmAvd_Tp-Z
- Some notes on usage: https://docs.google.com/document/d/1GbqYFBsTZWKoXu2yFs49sP9X311q8BsnjNsfHYYN_Wg/edit?tab=t.0
- If you are using EEG data specifically (@Antho2422) this may be helpful: https://github.com/glchau/TOTEM_for_EEG_code

**Comment by konsgken** [comment]
Hi,

I’ve been reviewing the TOTEM repo and have a question regarding its applicability to audio time series data. From what I understood TOTEM is trained on low-sampling rate time series, so can the pretrained model be used for high-sampling rate audio data (e.g., 20 KHz)?

Thanks in advance!

**Comment by SaberaTalukder** [comment]
We have trained TOTEM on audio data; it's pretty good but not as good (at least to my ear) as highly specialized audio models (but there are other tradeoffs to consider [e.g. training time]). There are easy hyperparameter tweaks that make TOTEM better for audio (e.g. change the compression factor). If you're going to use TOTEM for audio, I'd recommend tweaking hyperparams and training your own.

P.S. the neuroscience data does have a high sampling rate (one sample every 0.002 sec), so not as high as audio, but we did successfully train TOTEM on higher frequency signals.

**Comment by nvmaditya** [comment]
Hi,
From past week I have been trying to run TOTEM, I am new to deep learning and from what I understood is that the weights in the google drive are for the vqvae and the weights for the transformer are not available?

I am new and I am still learning, I would really appreciate if someone could help me run TOTEM for forecasting.

**Comment by SaberaTalukder** [comment]
tl;dr: the checkpoints for the transformer forecaster were lost, so I can't upload them. Especially on the smaller datasets it should not take too long to train your own. This is the code for the [transformer](https://github.com/SaberaTalukder/TOTEM/blob/master/forecasting/lib/models/decode.py); all the [scripts](https://github.com/SaberaTalukder/TOTEM/tree/master/forecasting/scripts) show how this code is called.

Check out this [thread](https://github.com/SaberaTalukder/TOTEM/issues/4) for slightly more detail.
